astropy/io/ascii/fastbasic.py

```diff
@@ -58,6 +58,53 @@ def _read_header(self):
         # can be overridden for specialized headers
         self.engine.read_header()
 
+    def _get_col_type(self, col_dtype):
+        # get the column type as Type classes defined in core
+        col_dtype = str(col_dtype)
+        if 'int' in col_dtype:
+            col_type = core.IntType
+        elif 'float' in col_dtype:
+            col_type = core.FloatType
+        elif 'bool' in col_dtype:
+            col_type = core.BoolType
+        elif 'str' in col_dtype:
+            col_type = core.StrType
+        else:
+            col_type = core.AllType
+        return col_type
+        
+    def _convert_vals(self, data):
+        converter_first = None
+        modified_col = False
+        
+        if self.converters is not None:
+            for col_name in data:
+                converters = self.converters.get(col_name, None) 
+                if converters is not None:
+                    col_dtype = data[col_name].dtype
+                    col_type = self._get_col_type(col_dtype)
+                    if len(converters) == 0:
+                        raise ValueError('Column {} failed to convert: no converters defined'.format(col_name))
+                    for converter in converters:
+                        try:
+                            converter_func , converter_type = converters
+                            if not issubclass(converter_type, core.NoType):
+                                raise ValueError()
```

Comment:
I've added the error message for this in the except block. Wouldn't that be fine?